NBitcoin.Altcoins/Stratis.cs

```diff
@@ -0,0 +1,660 @@
+				LockTime lockTimeTemp = LockTime.Zero;
+				stream.ReadWriteStruct(ref lockTimeTemp);
+
+				this.Version = nVersionTemp;
+				this.Time = nTimeTemp; // POS Timestamp
+				vinTemp.ForEach(i => this.AddInput(i));
+				voutTemp.ForEach(i => this.AddOutput(i));
+				this.LockTime = lockTimeTemp;
+			}
+
+			private void SerializeTxn(BitcoinStream stream, bool witSupported)
+			{
+				byte flags = 0;
+				var version = (witSupported && (this.Inputs.Count == 0 && this.Outputs.Count > 0)) ? this.Version | NoDummyInput : this.Version;
+				stream.ReadWrite(ref version);
+
+				// POS Timestamp
+				var time = this.Time;
+				stream.ReadWrite(ref time);
+
+				if (witSupported)
+				{
+					/* Check whether witnesses need to be serialized. */
+					if (HasWitness)
+					{
+						flags |= 1;
+					}
+				}
+				if (flags != 0)
+				{
+					/* Use extended format in case witnesses are to be serialized. */
+					TxInList vinDummy = new TxInList();
+					stream.ReadWrite<TxInList, TxIn>(ref vinDummy);
+					stream.ReadWrite(ref flags);
+				}
+				TxInList vin = this.Inputs;
+				stream.ReadWrite<TxInList, TxIn>(ref vin);
+				vin.Transaction = this;
+				TxOutList vout = this.Outputs;
+				stream.ReadWrite<TxOutList, TxOut>(ref vout);
+				vout.Transaction = this;
+				if ((flags & 1) != 0)
+				{
+					StratisWitness wit = new StratisWitness(this.Inputs);
+					wit.ReadWrite(stream);
+				}
+				LockTime lockTime = this.LockTime;
+				stream.ReadWriteStruct(ref lockTime);
+			}
+
+			public static StratisTransaction ParseJson(string tx)
+			{
+				JObject obj = JObject.Parse(tx);
+				StratisTransaction stratTx = new StratisTransaction(Stratis.StratisConsensusFactory.Instance);
+				DeserializeFromJson(obj, ref stratTx);
+
+				return stratTx;
+			}
+
+			private static void DeserializeFromJson(JObject json, ref StratisTransaction tx)
+			{
+				tx.Version = (uint)json.GetValue("version");
+				tx.Time = (uint)json.GetValue("time");
+				tx.LockTime = (uint)json.GetValue("locktime");
+
+				var vin = (JArray)json.GetValue("vin");
+				for (int i = 0; i < vin.Count; i++)
+				{
+					var jsonIn = (JObject)vin[i];
+					var txin = new TxIn();
+					tx.Inputs.Add(txin);
+
+					var script = (JObject)jsonIn.GetValue("scriptSig");
+					if (script != null)
+					{
+						txin.ScriptSig = new Script(Encoders.Hex.DecodeData((string)script.GetValue("hex")));
+						txin.PrevOut.Hash = uint256.Parse((string)jsonIn.GetValue("txid"));
+						txin.PrevOut.N = (uint)jsonIn.GetValue("vout");
+					}
+					else
+					{
+						var coinbase = (string)jsonIn.GetValue("coinbase");
+						txin.ScriptSig = new Script(Encoders.Hex.DecodeData(coinbase));
+					}
+
+					txin.Sequence = (uint)jsonIn.GetValue("sequence");
+
+				}
+
+				var vout = (JArray)json.GetValue("vout");
+				for (int i = 0; i < vout.Count; i++)
+				{
+					var jsonOut = (JObject)vout[i];
+					var txout = new TxOut();
+					tx.Outputs.Add(txout);
+
+					var btc = (decimal)jsonOut.GetValue("value");
```

Comment:
If it is possible to avoid using decimal for representing values I would recommend to avoid it. Sooner or later there will be problems because of that.